### DISCUSSION TITLE
ERRORE TOKEN

### DISCUSSION THREAD

**Discussion by Annaanto** [fattureincloud/api#75] [Q&A] [answered]
Buongiorno.
Sto cercando di testare le chiamate tramite l'API REFERENCE, ma non riesco a fare le chiamate.
Ottengo  il seguente errore:

{
  "error": "invalid_token",
  "error_description": "There was an error decoding the token."
}

Il token è corretto e anche il companyID. 
Uso la autenticazione HTTP Bearer 
Grazie
Anna

**Comment by valmoz** [comment, marked as answer]
Buongiorno @Annaanto,
questo errore avviene quando il token è stato fornito ma è errato per qualche motivo.

Le chiederei di controllare di non aver copiato male il token, magari aggiungendo uno spazio all'inizio/fine o non copiando gli ultimi caratteri. Le chiederei anche di controllare che il token inizi con il prefisso "a/" (senza virgolette) come indicato [a questo link](https://developers.fattureincloud.it/docs/authentication/code-flow/vanilla-code#-token-types).

Le segnalo inoltre che l'API reference inserisce automaticamente il prefisso "Bearer " nell'header authentication, quindi non è necessario inserirlo manualmente.

Grazie
Mauro

**Reply by Annaanto** [reply]
Grazie, in effetti non avevo messo "a/" prima del token.
Ora funziona.
Una precisazione: questo prefisso nel token si usa solo quando si testa tramite le API REFERENCE, giusto?
Grazie,
Buona giornata,
Anna

**Reply by valmoz** [reply]
Buongiorno,
no. Il prefisso "a/" serve sempre, ma dovrebbe essere inviato in automatico da tutti i metodi di autenticazione a disposizione.

Il prefisso "Bearer " invece dipende dal mezzo che sta utilizzando per effettuare le chiamate (ad esempio, l'API Reference e le SDK lo aggiungono in automatico, mentre se utilizza direttamente un HTTP Client sarà necessario inserirlo a mano)

Grazie
Mauro

**Reply by Annaanto** [reply]
Ah ok. 
In realtà nella chiamata che ho fatto dal nostro gestionale non l'ho messo e funziona ugualmente.
Probabilmente viene aggiunto in automatico.
Grazie,
Anna
